controllers/theatre.controller.js

```diff
@@ -80,9 +78,31 @@ exports.deleteTheatre = async (req, res) => {
 }
 
 exports.addMoviesToATheater = async (req, res) => {
-
+    const existingTheatre = await theatreModel.findOne({ theatreId: req.params.theatreId })
+    movieIds = req.body.movieIds
+    //Add movie ids to the theatre
+    if (req.body.insert) {
+        movieIds.forEach(movieId => {
+            existingTheatre.movies.push(movieId)
+        });
+    } else {
+        existingMovieIds = existingTheatre.movies
+        movieIds.forEach(movieId => {
+            existingMovieIds = existingMovieIds.filter(smi => smi != movieId)
+        })
+        existingTheatre.movies = existingMovieIds
+    }
+    await existingTheatre.save()
+    res.status(200).send(existingTheatre)
 }
 
 exports.checkMovieInsideATheatre = async (req, res) => {
+    const existingTheatre = await theatreModel.findOne({ theatreId: req.params.theatreId })
```

Comment:
wrong logic, check for movie in the theatre document. Hint: modify the query object